test/nodes/test_file_converter.py

```diff
@@ -422,6 +422,34 @@ def test_csv_to_document_with_wrong_qa_headers(tmp_path):
         node.run(file_paths=csv_path)
 
 
+@pytest.mark.unit
+def test_csv_to_document_with_wrong_qa_headers_raise_on_failure_true(tmp_path):
+    node = CsvTextConverter()
+    csv_path = tmp_path / "csv_qa_with_wrong_headers.csv"
+    rows = [
+        ["wrong", "headers"],
+        ["What is Haystack ?", "Haystack is an NLP Framework to use transformers in your Applications."],
+    ]
+    write_as_csv(rows, csv_path)
+
+    with pytest.raises(ValueError, match="The CSV must contain two columns named 'question' and 'answer'"):
+        node.run(file_paths=csv_path, raise_on_failure=True)
```

Comment:
```suggestion
    with pytest.raises(ValueError):
        node.run(file_paths=csv_path, raise_on_failure=True)
```

We prefer not to match the error message